### PR TITLE
Adds tests for not calling Stream.Flush and calling Stream.FlushAsync…

### PR DESCRIPTION
**Pull request by MihaMarkic**
Adds tests for not calling Stream.Flush and calling Stream.FlushAsync instead on an async XmlWriter #37457

### CODE REVIEW

File: src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Text;
```
Comment (Youssef1313) [inline src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:6]: Better to make them ordered alphabetically.

File: src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                FlushAsyncCalled = true;
+                return base.FlushAsync(cancellationToken);
+            }
+        }
+
+        [Fact]
+        public void AsyncWriter_DoesNotCall_Flush_On_Close()
+        {
+            var stream = new ExtendedMemoryStream();
+            var writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
```
Comment (Youssef1313) [inline src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:35]:
```suggestion
            XmlWriter writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
```

File: src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            writer.Close();
+
+            Assert.False(stream.FlushCalled);
+        }
+
+        [Fact]
+        public void AsyncWriter_DoesCall_FlushAsync_On_Close()
+        {
+            var stream = new ExtendedMemoryStream();
+            var writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
```
Comment (Youssef1313) [inline src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:45]:
```suggestion
            XmlWriter writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

**Comment by krwq** [comment]
Hi @MihaMarkic, we are currently during the repo merge (coreclr+corefx will become runtime repo) and this PR cannot be merged at this moment. Expect dotnet/runtime repository to show up in some time (unfortunately we don't have any official ETA but no longer than few weeks) where equivalent PR can be created (all files will have very similar structure so PR should be easily movable)

**Comment by MihaMarkic** [comment]
Hi @krwq 
So there will be a new repo then. Sure, no problem, I'll create a new PR once the repo is up. If you can, please let me know when the repo is up (but I guess I'll notice it anyway).

File: src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs (hunk 3 above)
Comment (MihaMarkic) [reply src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:45]: Why would you declare it explicitly as it is implied from the name and the statement, plus it's a small method.

Comment (Youssef1313) [reply src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:45]: @MihaMarkic, That's the corefx style-guide. It uses `var` when the type is clearly obvious from the right hand side. Clearly obvious means that it's determined by a language rule, like creating on object with: `Type type = new Type(..)` or casting.

**Comment by danmoseley** [comment]
As far as I know we're still on schedule to open the repo up next week
https://github.com/dotnet/coreclr/issues/27549

File: src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs (hunk 3 above)
Comment (danmoseley) [reply src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:45]: The style allows var pretty much only for "new" or an explicit cast or some explicit initializer like a constant string.
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

Comment (MihaMarkic) [reply src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:45]: OK, thanks guys

File: src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
```
Comment (krwq) [inline src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs:1]: copyright header missing

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in #27549 the dotnet/runtime repository will be used going forward for changes to this code base. Closing this PR as no more changes will be accepted into master for this repository. If you’d like to continue working on this change please move it to dotnet/runtime.

**Pull request closed without merging**